Feat: Dropdown multi-select support

morrisallison opened the pull request:
## Overview

* Added the `multiple` prop to the `Dropdown` component
    - When `true`, the `MultiDownshift` behavior is used instead of `Downshift`
* Added `@hig/multi-downshift`
    - Provides multi-selection support via Downshift
* Added `@hig/utils`
    - The package serves the purpose described in #768 (see _"Utilities package" section_).
* Upgraded to Downshift v2
    - Improved a11y for the menu
* Removed the disabled dropdown story per our discussion about story variants
* Fixed an issue with `undefined` prop-types in `InputPresenter.propTypes`

## Related

* Resolves #1001
* [Storybook](http://hig-feat-dropdown-multiselect.surge.sh/?selectedKind=Forms|Dropdown)

halfghaninne commented:
@morrisallison When @arirusso and I took a look at the component in Storybook, we noticed some unexpected behavior. We select three options from the multi-select, and then try to deselect the first one. It disappears from the list of selected options above, but remains checked in the list of options in the dropdown.

We can grab a screen recording if that helps!

morrisallison commented:
@halfghaninne No need, I see it now, thanks for finding it

morrisallison commented:
I added the fixes I was working on separately to this branch and moved the packages to stable release. The overall UX of the dropdown should be much better than before, and multiple selection should now work as expected.

nfiniteset commented:
The Dropdown on storybook.hig.autodesk.com shows a checkmark for the selected item, so this is a new bug.
